android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt

```diff
@@ -113,7 +113,7 @@ class AppLinksInterceptor(
         val redirect = useCases.interceptedAppLinkRedirect(uri)
         val result = handleRedirect(redirect, uri)
 
-        if (inUserDoNotIntercept(uri, redirect.appIntent)) {
+        if (engineSupportedSchemes.contains(uriScheme) && inUserDoNotIntercept(uri, redirect.appIntent)) {
```

Comment:
Just looking at the comments on the bug, do we still want to do `inUserDoNotIntercept` at all?

Reply:
yes, but we don't want to not intercept any scheme our engine doesn't support.  Example here is if the user tap on tel:/ link and tapped cancel, if they tap on it again, we should still intercept.  Since our browser doesn't support that link, if we don't intercept, nothing happens.